src/javascript/gameInstance.js

```diff
@@ -120,12 +124,15 @@ export default class GameInstance {
     this.endX = startX + (trackWidth + 1) * this.dropTrackArr.length - 1;
 
     const isMobile = window.innerWidth < 1000;
-    const hitLineProp = isMobile ? 8.5 : 9;
+    let hitLineProp = isMobile ? 8.5 : 9;
+    if (!this.vm.playMode) hitLineProp = this.vm.options.lowerHitLine ? 4 : 0;
 
-    this.vm.checkHitLineY = (this.canvas.height / 10) * hitLineProp;
-    this.vm.noteSpeedPxPerSec =
-      this.vm.checkHitLineY / Number(this.vm.noteSpeedInSec);
+    this.checkHitLineY = (this.canvas.height / 10) * hitLineProp;
+    this.noteSpeedPxPerSec = 380 * Number(this.vm.noteSpeed);
```

Comment:
Is this `Number(this.vm.noteSpeed)` cast really necessary? Also `Number(this.vm.noteSpeedInSec)` as well.

Reply:
I think I can remove it safely, will do that now